dialer.ts

```diff
@@ -0,0 +1,65 @@
+import { existsSync } from "fs";
+import { resolve } from "path";
+
+type Dialer = (context: any) => Promise<void> | void;
+
+class DialerRegistry {
+  private preDialers: Dialer[] = [];
+  private postDialers: Dialer[] = [];
+
+  registerPreDialer(dialer: Dialer) {
+    this.preDialers.push(dialer);
+  }
+
+  registerPostDialer(dialer: Dialer) {
+    this.postDialers.push(dialer);
+  }
+
+  async executePreDialers(context: any) {
+    if (this.preDialers.length === 0) {
+      return;
+    }
+
+    for (const dial of this.preDialers) {
+      await dial(context);
+    }
+  }
+
+  async executePostDialers(context: any) {
+    if (this.postDialers.length === 0) {
+      return;
+    }
+
+    for (const dial of this.postDialers) {
+      await dial(context);
+    }
+  }
+}
+
+export const dialerRegistry = new DialerRegistry();
+
+export async function registerDialers(dialPath: string) {
+  const resolvedDialPath = resolve(dialPath);
+
+  if (!existsSync(resolvedDialPath)) {
+    console.error(`Error: Dialer file not found: ${resolvedDialPath}`);
+    process.exit(1);
+  }
+
+  try {
+    const userModule = await import(resolvedDialPath);
+
+    Object.entries(userModule).forEach(([key, fn]) => {
+      if (typeof fn === "function") {
+        if (key.startsWith("pre")) {
+          dialerRegistry.registerPreDialer(fn as Dialer);
+        } else if (key.startsWith("post")) {
+          dialerRegistry.registerPostDialer(fn as Dialer);
+        }
+      }
+    });
+  } catch (error) {
+    console.error(`Failed to load dialers:`, error);
+    process.exit(1);
```

Comment:
Not sure if it should exit `rv`. Perhaps `rv` should run without calling any of the registered functions.

Reply:
This a good point. Agreed. Let's make sure it can work well with #86. (I think it will, as it won't be even the case.)